packages/webdriverio/src/commands/element/click.js

```diff
@@ -46,56 +54,82 @@
     })
  * </example>
  *
  * @alias element.click
  * @uses protocol/element, protocol/elementIdClick, protocol/performActions, protocol/positionClick
  * @type action
- * @param {Object=} options object containing a property called `button` which can be set to 0 (left) 1 (middle) or 2 (right) button
+ * @param options Object (optional)
+ * @param options.button can be one of [0, "left", 1, "middle", 2, "right"] (optional)
+ * @param options.x Number (optional)
+ * @param options.y Number (optional)
  */
 
-export default async function click (options) {
-    let { button } = options || {}
+export default async function click(options) {
+    if (typeof options === 'undefined') {
+        return this.elementClick(this.elementId)
+    }
+
+    if (typeof options !== 'object' || Array.isArray(options)) {
+        throw new TypeError('Options must be an oject')
+    }
+
+    let {
+        button = 0,
+        x: xoffset = 0,
+        y: yoffset = 0
+    } = options || {}
+
+    if (
+        typeof xoffset !== 'number'
+        || typeof yoffset !== 'number'
+        || !Number.isInteger(xoffset)
+        || !Number.isInteger(yoffset)) {
+        throw new TypeError('Coördinates must be integers')
+    }
 
     if (button === 'left') {
         button = 0
     }
-
     if (button === 'middle') {
         button = 1
     }
-
     if (button === 'right') {
         button = 2
     }
-
-    if (typeof button === 'undefined') {
-        return this.elementClick(this.elementId)
+    if (![0, 1, 2].includes(button)) {
+        throw new Error('Button type not supported.')
     }
 
-    if ([0, 1, 2].includes(button)) {
-        if (this.isW3C) {
-            await this.performActions([{
-                type: 'pointer',
-                id: 'pointer1',
-                parameters: { pointerType: 'mouse' },
-                actions: [
-                    { type: 'pointerMove', origin: this, x: 0, y: 0 },
-                    { type: 'pointerDown', button },
-                    { type: 'pointerUp', button }
-                ]
-            }])
-
-            return this.releaseActions()
-        }
+    if (this.isW3C) {
+        await this.performActions([{
+            type: 'pointer',
+            id: 'pointer1',
+            parameters: {
+                pointerType: 'mouse'
+            },
+            actions: [{
+                type: 'pointerMove',
+                origin: this,
+                x: xoffset,
+                y: yoffset
+            }, {
+                type: 'pointerDown',
+                button
+            }, {
+                type: 'pointerUp',
+                button
+            }]
+        }])
 
-        await this.moveTo()
-        return this.positionClick(button)
+        return this.releaseActions()
     }
 
-    throw new Error('Button type not supported.')
+    const { width, height } = await this.getElementSize(this.elementId)
```

Comment:
shouldn't we use https://webdriver.io/docs/api/element/getSize.html here instead?

Reply:
not really because it is definitely not the w3c case here